### PR TITLE
Remove AES-like alt interface

### PR DESCRIPTION
**Pull request by tom-daubney-arm**
## Description

Remove AES and AES-like ALT interfaces.

This PR is made up of commits split out from #9226 in order to ease review.

## PR checklist

Please tick as appropriate and edit the reasons (e.g.: "backport: not needed because this is a new feature")

- [x] **changelog** Will be provided at end of PR series
- [x] **3.6 backport** not required
- [x] **2.28 backport** not required
- [x] **tests** existing tests suffice

## Notes for the submitter

Please refer to the [contributing guidelines](https://github.com/Mbed-TLS/mbedtls/blob/development/CONTRIBUTING.md), especially the
checklist for PR contributors.

Help make review efficient:
* Multiple simple commits
  - please structure your PR into a series of small commits, each of which does one thing
* Avoid force-push
  - please do not force-push to update your PR - just add new commit(s)
* See our [Guidelines for Contributors](https://mbed-tls.readthedocs.io/en/latest/reviews/review-for-contributors/) for more details about the review process.

### CODE REVIEW

**Review by ronald-cron-arm — CHANGES_REQUESTED** [review]

File: library/aes.c [hunk 1]
```diff
@@ -1918,8 +1910,8 @@ int mbedtls_aes_self_test(int verbose)
 
             /*
              * AES-192 is an optional feature that may be unavailable when
-             * there is an alternative underlying implementation i.e. when
-             * MBEDTLS_AES_ALT is defined.
+             * there is an alternative underlying implementation such as a
+             * PSA driver.
```
[the remark this answers is not in the document]
Comment (tom-daubney-arm) [reply library/aes.c:1914]: The only issue here is that both of those config options `MBEDTLS_AES_SETKEY_ENC_ALT` and `MBEDTLS_AES_SETKEY_DEC_ALT` get removed in the final PR of the series i.e. the last commit of #9226. So given this, shall I just remove the comment altogether?

File: tests/include/test/psa_crypto_helpers.h [hunk 2]
```diff
@@ -290,8 +290,7 @@ uint64_t mbedtls_test_parse_binary_string(data_t *bin_string);
  *  \param  alg             The AEAD algorithm.
  *  \param  nonce_length    The nonce length in number of bytes.
  */
-#if defined(MBEDTLS_GCM_ALT) || \
-    defined(MBEDTLS_PSA_ACCEL_ALG_GCM)
+#if defined(MBEDTLS_PSA_ACCEL_ALG_GCM)
```
[the remark this answers is not in the document]
Comment (tom-daubney-arm) [reply tests/include/test/psa_crypto_helpers.h:293]: Ooh nice catch, thanks!

File: library/aes.c (hunk 1 above)
[the remark this answers is not in the document]
Comment (ronald-cron-arm) [reply library/aes.c:1914]: I was thinking to change the comment here and remove it in the last PR.

[the remark this answers is not in the document]
Comment (tom-daubney-arm) [reply library/aes.c:1914]: Ok cool, can do it like that

File: library/aes.c [hunk 3]
```diff
@@ -1918,8 +1910,9 @@ int mbedtls_aes_self_test(int verbose)
 
             /*
              * AES-192 is an optional feature that may be unavailable when
-             * there is an alternative underlying implementation i.e. when
-             * MBEDTLS_AES_ALT is defined.
+             * there is an alternative underlying implementation i.e when
```
Comment (ronald-cron-arm) [inline library/aes.c:1913]: Please change the three other instances as well, lines 1978-1979 ...

**Review by ronald-cron-arm — CHANGES_REQUESTED** [review]

File: library/aes.c (hunk 3 above)
Comment (tom-daubney-arm) [reply library/aes.c:1913]: Whooops... sorry...

**Review by ronald-cron-arm — DISMISSED** [review]
LGTM, thanks.

**Review by bensze01 — APPROVED** [review]
LGTM, please rebase it to resolve the conflicts

**Comment by tom-daubney-arm** [comment]
Note: OpenCI failures appear to be timeouts.

**Review by ronald-cron-arm — APPROVED** [review]
I've checked the rebase, LGTM.

**Comment by tom-daubney-arm** [comment]
@ronald-cron-arm Rebased again without needing manual conflict resolution.

**Review by ronald-cron-arm — APPROVED** [review]
LGTM

**Review by bensze01 — DISMISSED** [review]
LGTM

**Review by ronald-cron-arm — APPROVED** [review]
Last rebase LGTM

**Review by bensze01 — APPROVED** [review]
LGTM

**Pull request merged**